File: src/api/schemas/documents.py
"""API 请求/响应 Pydantic 模型"""

from __future__ import annotations

from pydantic import BaseModel, Field


class UploadResponse(BaseModel):
    doc_id: str
    filename: str
    dataset_id: str | None = None
    status: str
    uploaded_at: str


class DocumentStatusResponse(BaseModel):
    doc_id: str
    filename: str
    status: str  # pending | processing | done | failed
    error_msg: str | None = None
    uploaded_at: str | None = None
    updated_at: str | None = None


class IngestRequest(BaseModel):
    doc_ids: list[str] = Field(..., min_length=1, description="待摄入的文档 ID 列表")


class IngestResult(BaseModel):
    doc_id: str
    filename: str
    status: str
    error_msg: str | None = None


class IngestResponse(BaseModel):
    results: list[IngestResult]


class DocumentListItem(BaseModel):
    doc_id: str
    filename: str
    status: str
    error_msg: str | None = None
    uploaded_at: str | None = None
    updated_at: str | None = None


class DocumentListResponse(BaseModel):
    total: int
    page: int
    size: int
    items: list[DocumentListItem]

File: web/src/api/documents.ts
import request from './request'
import axios from 'axios'

export interface UploadResponse {
  doc_id: string
  filename: string
  dataset_id: string | null
  status: string
  uploaded_at: string
}

export interface DocumentStatusResponse {
  doc_id: string
  filename: string
  status: 'pending' | 'processing' | 'done' | 'failed'
  error_msg: string | null
  uploaded_at: string | null
  updated_at: string | null
}

export interface IngestResult {
  doc_id: string
  filename: string
  status: string
  error_msg: string | null
}

export interface IngestResponse {
  results: IngestResult[]
}

export const uploadDocuments = (files: File[], datasetId?: string) => {
  const formData = new FormData()
  files.forEach((f) => formData.append('files', f))
  if (datasetId) formData.append('dataset_id', datasetId)
  return axios.post<any, UploadResponse[]>('/api/v1/documents', formData, {
    timeout: 120000,
  })
}

export const ingestDocuments = (docIds: string[]) =>
  request.post<any, IngestResponse>('/documents/ingest', { doc_ids: docIds })

export const getDocumentStatus = (docId: string) =>
  request.get<any, DocumentStatusResponse>(`/documents/${docId}/status`)

export const deleteDocument = (docId: string) =>
  request.delete<any, { message: string; doc_id: string }>(`/documents/${docId}`)

export interface DocumentListItem {
  doc_id: string
  filename: string
  status: 'pending' | 'processing' | 'done' | 'failed'
  error_msg: string | null
  uploaded_at: string | null
  updated_at: string | null
}

export interface DocumentListResponse {
  total: number
  page: number
  size: number
  items: DocumentListItem[]
}

export const listDocuments = (params?: { dataset_id?: string; page?: number; size?: number }) =>
  request.get<any, DocumentListResponse>('/documents', { params })
